realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1302,4 +1411,112 @@ public E findFirst() {
             return null;
         }
     }
+
+    /**
+     * Represents a pending asynchronous Realm query.
+     *
+     * Users are responsible of maintaining a reference to {@code AsyncRequest} in order
+     * to call #cancel in case of a configuration change for example (to avoid memory leak, as the
+     * query will post the result to the caller's thread callback)
+     */
+    public class AsyncRequest {
+        private Future<?> pendingQuery;
+        private volatile boolean isCancelled = false;
+
+        public AsyncRequest(Future<?> pendingQuery) {
+            this.pendingQuery = pendingQuery;
+        }
+
+        /**
+         * Attempts to cancel execution of this query.
+         */
+        public void cancel () {
+            pendingQuery.cancel(true);
+            isCancelled = true;
+        }
+
+        private void setPendingQuery (Future<?> pendingQuery) {
+            this.pendingQuery = pendingQuery;
+        }
+
+        /**
+         * Whether an attempt to cancel the query was performed
+         * @return {@code true} if we already called {@link #cancel()} {@code false} otherwise
+         */
+        public boolean isCancelled() {
+            return isCancelled;
+        }
+    }
+
+    /**
+     * Custom {@link android.os.Handler} using the caller {@link android.os.Looper}
+     * to deliver result or error to a {@link io.realm.Realm.QueryCallback}
+     */
+    private class EventHandler extends Handler {
+        private final static String TABLE_VIEW_POINTER_ARG = "tvPtr";
+        private final static String CALLER_SHARED_GROUP_POINTER_ARG = "callerSgPtr";
+
+        private static final int MSG_SUCCESS = 1;
+        private static final int MSG_ERROR = MSG_SUCCESS + 1;
```

Comment:
it's just a style that helps preventing assigning the same value to a constant + defining a relative order between them. we should start adding those to our checkstyle/convention if we accept it or not